cxx/distributions/base.hh

```diff
@@ -18,6 +19,9 @@ class Distribution {
   // have been previously passed to incorporate().
   virtual void unincorporate(const T& x) {
     incorporate(x, -1.0);
+    // TODO: Debug why this fails sometimes, e.g. for the bigram_string
+    // emission. 
+    // assert(N >= 0);
```

Comment:
This is serious enough that we should probably create an issue for it.  (So that Alex etc. know about it when they revisit this codebase.)